### ISSUE TITLE
UART Support

### ISSUE THREAD

**Issue by esden** [gauteh/notecard-rs#3]
Are there any plans to also support UART as a protocol with this library?

I need to use the notecard over serial, and might be able to contribute some effort towards supporting UART.

If you are planning to add it, do you have specific thoughts on how you would like it done?

**Comment by gauteh** [comment]
Yeah that would be great! Maybe the serial::Write trait in embedded hal is shared between i2c and uart? And it just need some generalization?

Blues also added a binary interface for larger data transfer that I would like to implement. And the interface would actually some times be more flexible and robust with allocation. Or at least truncating heapless strings (there's an open issue on that on heapless).

**Comment by gauteh** [comment]
I had a stupid bug where the driver failed in places with long names, because I had chosen a too short string length.

**Comment by esden** [comment]
Here is one way of implementing this. https://github.com/analogdevicesinc/adi-adxl355-rs/blob/main/adi-adxl355/src/lib.rs
But maybe there is a better and cleaner solution.

**Comment by gauteh** [comment]
Yes, I think in practice you have to do something like that, because the differences are so big that you can't just swap the Write trait (e.g. https://github.com/gauteh/notecard-rs/blob/main/src/lib.rs#L291 or https://dev.blues.io/guides-and-tutorials/notecard-guides/serial-over-i2c-protocol/). Wheras with UART it is probably easier, you don't have to send and allocate the number of bytes to read, but you read until the response is complete?
